src/anomalib/trainer/utils/normalizer.py

```diff
@@ -0,0 +1,137 @@
+
+    This is responsible for updating the normalization values and normalizing the outputs.
+
+    Args:
+        normalization_method (NormalizationMethod): Normalization method. Defaults to None
+    """
+
+    def __init__(
+        self, trainer: core.AnomalibTrainer, normalization_method: NormalizationMethod = NormalizationMethod.NONE
+    ):
+        self.normalization_method: NormalizationMethod = normalization_method
+        self.trainer = trainer
+
+    @property
+    def normalization_metrics(self) -> Metric:
+        """Returns normalization metrics.
+
+        Checks if the trainer has anomaly module. If the trainer does, it searches for the normalization metrics in the
+        anomaly module. If it does not find it, it assigns the normalization metrics.
+        """
+        if self.trainer.lightning_module is not None:
+            if not hasattr(self.trainer.lightning_module, "normalization_metrics"):
+                setattr(self.trainer.lightning_module, "normalization_metrics", self._assign_normalization_metrics())
+            return self.trainer.lightning_module.normalization_metrics.cpu()
+        else:
+            raise ValueError("Trainer does not have a lightning module assigned.")
+
+    def _assign_normalization_metrics(self) -> Metric | None:
+        """Assign normalization metrics."""
+        # TODO change logic here
+        normalization_metrics: Metric | None = None
+        if self.normalization_method != NormalizationMethod.NONE:
+            if self.normalization_method == NormalizationMethod.MIN_MAX:
+                normalization_metrics = MinMax().cpu()
+            elif self.normalization_method == NormalizationMethod.CDF:
+                # TODO CDF only works for padim and stfpm. Check condition
+                # TODO throw error if nncf optimization is enabled.
+                normalization_metrics = AnomalyScoreDistribution().cpu()
+            else:
+                raise ValueError(f"Normalization method {self.normalization_method} is not supported.")
+        return normalization_metrics
+
+    def update_metrics(self, outputs: STEP_OUTPUT):
+        """Update values
+
+        Args:
+            outputs (STEP_OUTPUT): Outputs used for gathering normalization metrics.
+        """
+        if self.normalization_method == NormalizationMethod.MIN_MAX:
+            if "anomaly_maps" in outputs:
+                self.normalization_metrics(outputs["anomaly_maps"])
+            elif "box_scores" in outputs:
+                self.normalization_metrics(torch.cat(outputs["box_scores"]))
+            elif "pred_scores" in outputs:
+                self.normalization_metrics(outputs["pred_scores"])
+            else:
+                raise ValueError(
+                    "No values found for normalization, provide anomaly maps, bbox scores, or image scores"
+                )
+        elif self.normalization_method == NormalizationMethod.CDF:
+            self._standardize_batch(outputs)
```

Comment:
Not sure if this class should hold normalization-method-specific logic.

Reply:
Should I add methods such as `normalize_batch` inside `post_processing/normalization/cdf.py`. I can then directly call those methods here. Not sure if that's the right place for this.

Reply:
Not sure what would be the best solution here. It's just that I don't like seeing normalization method specific logic in the normalizer. Because this would mean that adding a new normalization type could involve changing the normalizer. Ideally this should not be necessary. We could consider addressing this separately to speed things up.